### DISCUSSION TITLE
How to solve the problem of the mismatch between ini file and mask map during calibrating process?

### DISCUSSION THREAD

**Discussion by jane-627** [iiasa/CWatM#17] [Q&A] [answered]
Hi, I got a serious problem and I need your help!
I ran the simulation process on Brahmaputra Basin smoothly, setting:
stepstart=1/1/1960
spinup=1/1/1970
stepend=31/12/1990
I save the initial file on 31/12/1969
and I ran the calibration process setting
stepstart=1/1/1970
spinup=None
stepend=31/12/1990
and I use the initial file saved before. and I got a problem called:
b'\r\n(py36) D:\\CWatM\\calibration\\catchments\\00_001>python D:\\CWatM\\source_py3\\cwatm3.py settings_Brahmaputra_cali-Run00_001.ini -v \r\n\r\n\r\n ========================== CWATM ERROR =============================\r\n===== Problem reading initial data ====== \r\nInitial value: forest_w1 is has not the same shape as the mask map\r\nMaybe put"load_initial = False"\r\n\r\n'

But I have checked the size of areamap and initial file, they are all 189grids*90grids, what's the problem?
[source_py3.zip](https://github.com/iiasa/CWatM/files/6567356/source_py3.zip)
[calibration.zip](https://github.com/iiasa/CWatM/files/6567361/calibration.zip)

[Brahmaputra_5min.zip](https://github.com/iiasa/CWatM/files/6567372/Brahmaputra_5min.zip)
[ini_19691231.zip](https://github.com/iiasa/CWatM/files/6567375/ini_19691231.zip)

Jane

**Comment by guillaumot** [comment]
Hi Jane,
Happy to help you. Did you change something between your initial run and the calibration run?
Perhaps you already try, but it could be that the landcover fraction in 1969 is different than in 1970. What if you put stepend=01/01/1970 instead stepend=31/12/1969 in your intialization run?

Luca

**Comment by PeterBurek** [comment]
Hi Jane,
your areamap Brahmaputra_5min.tif is shifted. It does not fit to all the other maps
It start with top latitude : 31.03212..  but it should fit to all the other map:
I created another area ma with starts at 31.25
Maybe load both tifs in a GIS prg and you see what I mean.

We uploaded a new version and I put a settings file for Brahma in. Hopefully you only have to change the path names and it should run.

[brahma.zip](https://github.com/iiasa/CWatM/files/6663459/brahma.zip)

I tested also your Brahmaputra_5min.ti. It runs and produce init maps, but i got the same error when loading them again.
But it works if you use my areamap brahma.tif

Cheers,
Peter

**Reply by jane-627** [reply]
Thank you for your advice, but I'm afraid that it doesn't work. I've sent you an e-mail for detail problem and data. I wonder if there is some problem with my python version or my input data. Could this be possible for CWatM v1.05 running under anaconda py36 environment? Or where could I find the updated input data set(cwat_input.zip for the 30’ global version and cwat_input5min.zip for the 5’ global version mentioned on https://cwatm.iiasa.ac.at/)? Thank you very much!!

**Reply by guillaumot** [reply]
But you managed to run the model once to save your initial map, right? So, the Python version and input data should be ok.
You can test the model  and input maps by running:
"python cwatmpath.py settings.ini -c "

Let us know!
Best, Luca

**Reply by jane-627** [reply]
I found the problem that is in the calibration_single2_Py37.py. Although I can generate discarge_monthavg.tss during calibration process, but the command line shows no simulated streamflow is found. So I check the calibration_single2_Py37.py and found os.path.isfile(Qsim_tss) is always False, but Qsim_tss can be printed as its full path, and the type of Qsim_tss is string. I don't know how to solve the problem, could you please help me?
[calibration_single2_Py37.zip](https://github.com/iiasa/CWatM/files/6679900/calibration_single2_Py37.zip)

**Reply by guillaumot** [reply]
Hi Jane,
thanks for the details. You have this line just before:
Qsim_tss = os.path.join(directory_run, dischargetss)
I guess you already checked that Qsim_tss corresponds to what you want in term of string.
Could you check "directory_run" like this: os.path.isdir(directory_run) ?
Is there any specific character in your Qsim_tss that could create an error?
Or try os.path.isfile(Qsim_tss) by replacing Qsim_tss manually by your file pathname
Luca

**Reply by jane-627** [reply]
I guess I find the error. The time period I wrote in the settings.txt (Monthly) was not consistent with the time period in calibration python file(monthly). And now the calibration program can run smoothly on my computer. I'm very appreciated for your and Peter's help. Thank you very much!

**Reply by guillaumot** [reply]
Great!

**Comment by PeterBurek** [comment, marked as answer]
Btw there is a possibilty to generate a mask map with CWatM (see also tutorial example 5)

on the first run you to put the last point of your basin in maskmap (here for Brahmaputra)
MaskMap = 89.7 23.86

and you  add a line at the end of the settings file:
OUT_MAP_TotalEnd = catchment

Then you run CWatM once and this will produce a file: catchment_totalend.nc in the output folder.
(it automatically generates the basin, based on the network map)

This file is your new maskmap. You can either copy it and use it directly or convert it to a .tif file
Put it in the settingsfile and run:
MaskMap = E:/brahma/catchment_totalend.nc

For sure you can also rename it.
Do not forget to delete the last line  OUT_MAP_TotalEnd = catchment. It is not needed anymore.

**Reply by jane-627** [reply]
And I also have a question about how to run the calibration process on a computer of linux system, since the manual only introduce the compiling process for simulation under linux environment. Is it even possible?

**Reply by guillaumot** [reply]
Hi Jane,
yes you can run the calibration on Windows or Linux. You just have to take care of pathnames.

Luca
